pkg/server/interfaces/api/repository.go

```diff
@@ -131,6 +133,28 @@ func (h repository) GetWebServiceRoute() *restful.WebService {
 		Returns(400, "Bad Request", bcode.Bcode{}).
 		Writes([]string{}))
 
+	ws.Route(ws.POST("/registrysecrets/verify").To(h.verifyRepositorySecret).
+		Doc("Verify image repository secret.").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Filter(h.RbacService.CheckPerm("project/config", "list")).
+		Param(ws.QueryParameter("project", "the config project").DataType("string").Required(true)).
+		Param(ws.QueryParameter("name", "the config").DataType("string").Required(true)).
+		Reads(v1.CreateConfigRequest{}).
+		Returns(200, "OK", v1.ValidateRepoResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(v1.ValidateRepoResponse{}))
+
+	ws.Route(ws.GET("/repositorytags").To(h.getRepositoryTags).
```

Comment:
This API is not suited to all image registries, right?

Reply:
DockerHub and Harbor have been verified